httpx/_content.py

```diff
@@ -150,11 +151,21 @@ def encode_urlencoded_data(
 
 
 def encode_multipart_data(
-    data: dict, files: RequestFiles, boundary: Optional[bytes] = None
+    data: dict,
+    files: RequestFiles,
+    boundary: Optional[bytes] = None,
+    headers: Optional[Mapping[str, str]] = None,
 ) -> Tuple[Dict[str, str], MultipartStream]:
+    if headers and boundary is None and "content-type" in headers:
+        content_type = headers["content-type"]
+        if not content_type.startswith("multipart/form-data"):
```

Comment:
I think we could drop this case, as with [this comment[(https://github.com/encode/httpx/pull/2278/files#r905947972)

Reply:
done